assets/js/common/OperationModals/SaptuneSolutionOperationModal.jsx

```diff
@@ -54,25 +56,39 @@ function SaptuneSolutionOperationModal({
   const [checked, setChecked] = useState(false);
   const [solution, setSolution] = useState(NOT_SELECTED);
 
-  const availableSolutions = filter(solutions, ({ available }) =>
-    available(isHanaRunning, isAppRunning)
-  );
+  const availableSolutions = pipe(
+    filter(({ available }) => available(isHanaRunning, isAppRunning)),
+    map((solutionOption) => ({
+      ...solutionOption,
+      disabled: solutionOption.value === currentlyApplied,
+    }))
+  )(solutions);
+
+  const resetSolution = () => setSolution(currentlyApplied || NOT_SELECTED);
+
+  useEffect(() => {
+    setChecked(false);
+    resetSolution();
+  }, [currentlyApplied]);
+
+  const resetState = () => {
+    resetSolution();
+    setChecked(false);
+  };
 
   return (
     <OperationModal
       title={title}
       description="Select Saptune tuning solution"
       operationText="Saptune solution"
-      applyDisabled={!checked || solution === NOT_SELECTED}
+      applyDisabled={
+        !checked || solution === NOT_SELECTED || solution === currentlyApplied
```

Comment:
Do we really need the `solution === currentlyApplied`?
You cannot never select the current solution, it is disabled